other/lru_cache.py

```diff
@@ -114,11 +132,46 @@ def set(self, key: int, value: int) -> None:
             node.val = value
             self.list.add(node)
 
-    def has_key(self, key: int) -> bool:
+    def cache_info(self) -> str:
         '''
-        Checks if the input key is present in cache
+        Returns the details for the cache instance
+        [hits, misses, capacity, current size]
         '''
-        return key in self.cache
+
+        return f'CacheInfo(hits={self.hits}, misses={self.miss}, \
+capacity={self.capacity}, current size={self.num_keys})'
+
+    @staticmethod
+    def decorator(size: int = 128):
+        '''
+        Decorator version of LRU Cache
+        '''
+
+        def cache_decorator_inner(func: Callable):
+
+            def cache_decorator_wrapper(*args, **kwargs):
+                if func not in LruCache.decorator_function_to_instance_map:
+                    LruCache.decorator_function_to_instance_map[func] = LruCache(size)
+
+                result = LruCache.decorator_function_to_instance_map[func].get(args[0])
+
+                if result is not None:
+                    return result
+
+                result = func(*args, **kwargs)
+                LruCache.decorator_function_to_instance_map[func].set(args[0], result)
+                return result
+
+            def cache_info():
+                if func not in LruCache.decorator_function_to_instance_map:
+                    return "Cache for function not initialized"
```

Comment:
Should this raise an Exception?

Reply:
Yes, but later realized, its an impossible case, so removed it